worlds/Files.py

```diff
@@ -135,31 +126,58 @@ def read(self, file: Optional[Union[str, BinaryIO]] = None) -> None:
                         message = f"{arg0} - "
                 raise InvalidDataError(f"{message}This might be the incorrect world version for this file") from e
 
-    def read_contents(self, opened_zipfile: zipfile.ZipFile) -> None:
+    def read_contents(self, opened_zipfile: zipfile.ZipFile) -> Dict[str, Any]:
```

Comment:
This is something I didn't notice before.
related to https://github.com/ArchipelagoMW/Archipelago/pull/4516#discussion_r1921297275

Not only is this ambiguous to both mutate and return something, this is also a breaking change.
If this base class returns something in this method, every subclass that overrides it also needs to return the same type.
That could include apworlds that are not in this repo, as well as some in this repo.

If this changes here, it needs to change in every subclass, and we need to think about the deprecation for external apworlds.